tests/ClientBuilderTest.php

```diff
@@ -151,6 +151,26 @@ public function testGetClient(): void
         $this->assertAttributeSame($this->getObjectAttribute($clientBuilder, 'httpClient'), 'client', $httpClientPlugin);
     }
 
+    public function testClientHasDefaultIntegration(): void
```

Comment:
Rethinking about this test it seems to me that it's not as useful as it should since it's only checking that the list of integrations is empty/not empty but not whether it contains the right list of integrations. We should change the code to something like:

`$this->assertEquals($expectedIntegrations, $client->getOptions()->getIntegrations(), '', 0.0, 10, e)`

Reply:
What about `$this->assertEquals(\array_values($client->getOptions()->getIntegrations()), \array_values($integrations));` to not overcomplicate things?

Reply:
I agree that what I wrote is not user-friendly but we don't have named parameters in PHP unfortunately, so to use the last argument of the function `assertEquals` we must specify all the arguments that comes before too. By the way I wrote the code that way because that `boolean` parameter normalizes both the input and expected arrays by sorting them so that the test doesn't fail if their values are not in the same exact order. `array_values` instead returns all the values of an array reindexing but not sorting them. We should use `sort` on both arrays but at this point I would simply use the convenience of the `assertEquals` parameters to leave to PHPUnit the burden of normalization

Reply:
Maybe doing the asserts in a foreach is easier?

Reply:
We would still keep track of the fact that some keys may exists in an array but not in the other (so the test must fail) and viceversa. I don't think it's worth to create a custom code to do the assertion when PHPUnit already provides the way to do it even though it's not user friendly, we just have to write some readable code that uses its methods, e.g.

```php
$integrationsClassNames = array_map('get_class', $client->getOptions()->getIntegrations());

$this->assertEquals($expectedIntegrationsClassNames, $integrationsClassNames, '', 0, 10, true)
```